### ISSUE TITLE
Indicate *Why* refactoring a Worksheet name greater than 31 characters is prohibited

### ISSUE THREAD

**Issue by IvenBach** [rubberduck-vba/Rubberduck#5397]
**Justification**
Don't just prohibit worksheet names greater that 31 characters. Educate the user about the character limit.

**Description**
Not allowing a user to enter a long descriptive name without reason doesn't educate the user. Telling them the reason why they're limited does. Informed users code better. The red X positioned in the top right would benefit from either hovertext or a label displaying above it with *why* they can't do it.

**Comment by BZngr** [comment]
Agreed.  There are name validations with user messages are available from `VBAIdentifierValidator.TryMatchInvalidIdentifierCriteria(string name, out string criteriaMatchMessage)`.  The `criteriaMatchMessage` contains the feedback regarding the reason for a validation failure.  `VBAIdentifierValidator` is used in the `RefactoringRename` UI, but currently, the criteria messages are not surfaced to the user.
